src/Workleap.DomainEventPropagation.Subscription.PullDelivery/EventPullerService.cs

```diff
@@ -108,7 +109,7 @@ private async IAsyncEnumerable<EventGridClientAdapter.EventGridClientAdapter.Eve
                     continue;
                 }
 
-                var bundles = await this._eventGridTopicSubscription.Client.ReceiveCloudEventsAsync(this._eventGridTopicSubscription.TopicName, this._eventGridTopicSubscription.SubscriptionName, availableHandlers, cancellationToken).ConfigureAwait(false);
+                var bundles = await this._eventGridTopicSubscription.Client.ReceiveCloudEventsAsync(this._eventGridTopicSubscription.TopicName, this._eventGridTopicSubscription.SubscriptionName, Math.Min(availableHandlers, MaxEventRequestSize), cancellationToken).ConfigureAwait(false);
```

Comment:
BTW, don't we have the same issue for acknowledge/release/reject requests?

Reply:
The client accepts an `IEnumerable`, and this is what is supplied to it. I haven't found any limit in the documentation, so I would expect the client to batch the data properly.